ui/apps/dashboard/src/app/(organization-active)/(dashboard)/settings/integrations/vercel/projects.tsx

```diff
@@ -33,10 +31,8 @@ import { useVercelIntegration } from './useVercelIntegration';
 //   </AlertModal>
 // );
 
-export default function VercelProjects() {
-  const { data } = useVercelIntegration();
-  const router = useRouter();
-  const { projects } = data;
+export default function VercelProjects({ integration }: { integration: VercelIntegration }) {
+  const { projects } = integration;
```

Comment:
What happens to this data if a project is enabled and the user clicks back to this either via breadcrumbs or back button? Does this data refetch? Or do we need to invalidate some sort of cache for this view. I believe next might have changed this behavior also recently, so it may depend on our version.

Reply:
I think browser back will refetch and rerender, but docs are vague, I will research and find out.

Reply:
Routes prefetched by link prefetch directive remain in the cache for a duration of 5 minute by default so hitting the back button within 5 minutes will retrieve the route from cache. There are also several programatic escape hatches here using router.refresh or revalidatePath or revalidateTag that we can bind to navigation where data is more realtime.

https://nextjs.org/docs/app/building-your-application/caching#router-cache